src/transformers/video_utils.py

```diff
@@ -0,0 +1,828 @@
+    """
+    # It's possible the video has pixel values in [0, 255] but is of floating type
+    return np.min(video) >= 0 and np.max(video) <= 1
+
+
+def convert_pil_frames_to_video(videos: List[VideoInput]) -> List[Union["np.ndarray", "torch.Tensor"]]:
+    """
+    Given a batch of videos, converts each video to a 4D array. If video is already in array type,
+    it is simply returned. We assume that all inputs in the list are in the same format, based on the type of the first element.
+
+    Args:
+        videos (`VideoInput`):
+            Video inputs to turn into a list of videos.
+    """
+
+    if not isinstance(videos[0], (list, tuple)):
+        return videos
+
+    video_converted = []
+    for video in videos:
+        video = [np.array(frame) for frame in video]
+        video = np.stack(video)
+        video_converted.append(video)
+    return video_converted
+
+
+def make_batched_videos(videos) -> List[Union["np.ndarray", "torch.Tensor"]]:
+    """
+    Ensure that the input is a list of videos. If the input is a single video, it is converted to a list of length 1.
+    If the input is a batch of videos, it is converted to a list of 4D video arrays. Videos passed as list `PIL.Image`
+    frames are converted to 4D arrays.
+
+    We assume that all inputs in the list are in the same format, based on the type of the first element.
+
+    Args:
+        videos (`VideoInput`):
+            Video inputs to turn into a list of videos.
+    """
+    if not valid_videos:
+        raise ValueError(
+            f"Invalid video input. Expected either a list of video frames or an input of 4 or 5 dimensions, but got"
+            f" type {type(videos)}."
+        )
+
+    if is_batched_video(videos):
+        pass
+    elif is_valid_video(videos):
+        videos = [videos]
+    # only one frame passed, thus we unsqueeze time dim
+    elif is_valid_image(videos):
+        videos = [np.array(videos)[None, ...]]
+    # nested batch so we need to unflatten
+    elif isinstance(videos[0], (list, tuple)) and is_valid_video(videos[0][0]):
+        return [video for sublist in videos for video in sublist]
+    return convert_pil_frames_to_video(videos)
+
+
+def infer_channel_dimension_format(
+    video: np.ndarray, num_channels: Optional[Union[int, Tuple[int, ...]]] = None
+) -> ChannelDimension:
+    """
+    Infers the channel dimension format of `video`.
+
+    Args:
+        video (`np.ndarray`):
+            The video to infer the channel dimension of.
+        num_channels (`int` or `Tuple[int, ...]`, *optional*, defaults to `(1, 3)`):
+            The number of channels of the video.
+
+    Returns:
+        The channel dimension of the video.
+    """
+    num_channels = num_channels if num_channels is not None else (1, 3)
+    num_channels = (num_channels,) if isinstance(num_channels, int) else num_channels
+
+    if video.ndim == 4:
+        first_dim, last_dim = 1, 3
+    elif video.ndim == 5:
+        first_dim, last_dim = 2, 4
```

Comment:
the only diff with the same function for images, maybe just extend existing one? not feel strong re this